packages/apollo-datasource-rest/src/RESTDataSource.ts

```diff
@@ -104,7 +104,8 @@ export abstract class RESTDataSource<TContext = any> extends DataSource {
 
   protected parseBody(response: Response): Promise<object | string> {
     const contentType = response.headers.get('Content-Type');
-    if (contentType && contentType.startsWith('application/json')) {
+    const applicationJsonRegEx = /^application\/(\S*\+)?json/;
+    if (contentType && applicationJsonRegEx.test(contentType)) {
```

Comment:
Rather than leaning on this regular expression, can we explicitly enumerate the `Content-type`s which should be allowed (preferably outside the `RESTDataSource` as a `const` for performance since this code is executed on every request)?

Reply:
Unrelated to your change: I'm not sure why the existing implementation uses `startsWith`.  What `Content-type` starts with `application/json` but has something after it?  Any ideas?

Reply:
Not sure I made myself clear in the above comment.
I agree that the const should be defined outside ot RESTDataSource, but I think that a regular expression is better than a whitelist in the form of an enumeration.
What do you think?

Reply:
+1 For a specific list in this age of security.

Although, I'm not sure hard coding a broad list in source is ideal though, since some implementations wouldn't consume/respond with additional vendor types?

I may be misunderstanding.

Reply:
Realistically, I think this needs to be a configurable aspect of the DataSource which defaults to only `application/json` — rather than including any vendored `Content-type` that starts with `application/` and ends with `json`.

Reply:
I like the idea to make this a configurable aspect of the RESTDataSource. That would be the most flexible solution.

Reply:
@abernix I changed the PR according to our conversation over a month ago. I have not heard anything back.
What is happening?